src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,155 @@
+        }
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (checkIndexGreaterEqualSize(index) || checkForNegativeIndex(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Cannot set value by index: "
+                    + "Index is out of bounds");
+        } else {
+            elements[index] = value;
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (checkIndexGreaterEqualSize(index) || checkForNegativeIndex(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Cannot remove the element by index: "
+                    + "Index is out of bounds");
+        }
+        T removedElement = (T) elements[index];
+        int startCopyIndex = index + CHANGE_BY_ONE;
+        copyElements(elements, startCopyIndex,
+                    elements, index, size - startCopyIndex);
+        size--;
+        return removedElement;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        int index = ELEMENT_NOT_FOUND_INDEX;
+        index = findElement(element);
+        if (index != ELEMENT_NOT_FOUND_INDEX) {
+            return remove(index);
+        }
+        throw new NoSuchElementException("Element is not found in the List");
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == EMPTY_LIST_SIZE;
+    }
+
+    private void defaultGrowArrayIfFull() {
+        grow(elements.length + elements.length / DEFAULT_ARRAY_GROW_COEFFICIENT);
+    }
+
+    private void givenSizeGrowArrayIfFull(int necessarySize) {
+        grow(necessarySize);
+    }
+
+    private void grow(int size) {
+        Object[] enlargedElements = new Object[size];
+        copyElements(elements, ARRAY_BEGINNING_INDEX,
+                    enlargedElements, ARRAY_BEGINNING_INDEX, elements.length);
+        elements = enlargedElements;
+    }
+
+    private boolean checkIndexGreaterEqualSize(int index) {
+        return index >= size;
+    }
+
+    private boolean checkForNegativeIndex(int index) {
+        return index < ARRAY_BEGINNING_INDEX;
+    }
+
+    private void increaseSize(int value) {
+        size = size + value;
+    }
+
+    private int findElement(T element) {
+        for (int i = 0; i < elements.length; i++) {
+            if (elements[i] != null && elements[i].equals(element)
+                    || elements[i] == element) {
+                return i;
+            }
+        }
+        return ELEMENT_NOT_FOUND_INDEX;
+    }
+
+    private void copyElements(Object[] copyFromArray, int startIndexFromArray,
+                              Object[] copyToArray, int startIndexToArray,
+                              int numberOfElementsToCopy) {
+        System.arraycopy(copyFromArray, startIndexFromArray,
+                copyToArray, startIndexToArray, numberOfElementsToCopy);
     }
```

Comment:
you created separate method just to call other method with same arguments.
Just use  System.arraycopy in code, will be absolutely same